Makefile

```diff
@@ -35,60 +38,79 @@ docker-build-deps:
 
 .PHONY: docker-build-base
 docker-build-base:
-	docker build --build-arg REGISTRY=$(DOCKER_REGISTRY) --build-arg PRESIDIO_DEPS_LABEL=$(PRESIDIO_DEPS_LABEL) -t $(DOCKER_REGISTRY)/$(GOLANG_BASE) -f Dockerfile.golang.base .
+	-docker pull $(DOCKER_REGISTRY)/$(GOLANG_BASE):$(PRESIDIO_BRANCH_LABEL) || echo "\nCould not pull shared base Go image from registry, building locally. If you planned to build locally, the previous error message could be ignored\n"
+	docker build --build-arg REGISTRY=$(DOCKER_REGISTRY) --cache-from=$(DOCKER_REGISTRY)/$(GOLANG_BASE):$(PRESIDIO_BRANCH_LABEL) --build-arg PRESIDIO_DEPS_LABEL=$(PRESIDIO_DEPS_LABEL) -t $(DOCKER_REGISTRY)/$(GOLANG_BASE):$(PRESIDIO_LABEL) -f Dockerfile.golang.base .
 
 
 # To use docker-build, you need to have Docker installed and configured. You should also set
 # DOCKER_REGISTRY to your own personal registry if you are not pushing to the official upstream.
 .PHONY: docker-build
-docker-build: docker-build-base
-docker-build: $(addsuffix -dimage,$(IMAGES))
+docker-build: docker-build-golang
+docker-build: docker-build-python
+
+.PHONY: docker-build-golang
+docker-build-golang: docker-build-golang-base
+docker-build-golang: $(addsuffix -dimage,$(GOLANG_IMAGES))
+
+.PHONY: docker-build-golang-single
+docker-build-golang-single: $(addsuffix -dimage,$(GOLANG_IMAGE))
+
+.PHONY: docker-build-golang-base
+docker-build-golang-base: docker-build-base
+
+.PHONY: docker-build-python
+docker-build-python: $(addsuffix -dpypiimage,$(PYTHON_IMAGES))
+
 
 %-dimage:
-	docker build $(DOCKER_BUILD_FLAGS) --build-arg REGISTRY=$(DOCKER_REGISTRY) --build-arg VERSION=$(VERSION) --build-arg PRESIDIO_DEPS_LABEL=$(PRESIDIO_DEPS_LABEL) -t $(DOCKER_REGISTRY)/$*:$(PRESIDIO_LABEL) -f $*/Dockerfile .
+	docker build $(DOCKER_BUILD_FLAGS) --build-arg REGISTRY=$(DOCKER_REGISTRY) --build-arg VERSION=$(VERSION) --build-arg PRESIDIO_DEPS_LABEL=$(PRESIDIO_DEPS_LABEL) --build-arg PRESIDIO_BASE_LABEL=$(PRESIDIO_LABEL) -t $(DOCKER_REGISTRY)/$*:$(PRESIDIO_LABEL) -f $*/Dockerfile .
+
+%-dpypiimage:
+ifndef PIP_EXTRA_INDEX_URL
```

Comment:
Can you elaborate?
e.g. line 72 has wheel ver. line 70 doesnt. on purpose?

Reply:
its more than that. line 70 uses the local docker file. 72 uses Dockerfile which in turn, uses the wheel.
added comment in line.